### DISCUSSION TITLE
Mistral ai

### DISCUSSION THREAD

**Discussion by mh292929** [Taewan-P/gpt_mobile#89] [Ideas]
Hi. Please add Mistral ai model

**Comment by Taewan-P** [comment]
Hello,
For now, you can use the OpenAI platform instead, with custom url set to mistral's. I can see that they use OpenAI [specification](https://docs.mistral.ai/api/), so enter the api url to just before the `/v1/chat/completions`.

**Comment by mh292929** [comment]
I test it with https://api.mistral.ai/v1/chat/completions/ and model mistral-large-latest doesn't work

**Comment by Taewan-P** [comment]
Explain in more details. What did you put in API address? What kind of error does it show? If possible, do you see any logcat output?

**Comment by mh292929** [comment]
![Screenshot_۲۰۲۴۱۰۲۸-۱۷۳۵۱۶](https://github.com/user-attachments/assets/7d14241d-9e11-4959-aec5-6f0405f2d706)

I got unknown error I don't have log

**Comment by Taewan-P** [comment]
Remove `/v1/chat/completions/` part from your URL. It should be just `https://api.mistral.ai/`

**Comment by mh292929** [comment]
Thanks, fixed. Please add upload picture to this app

**Comment by 4kenhama** [comment]
Hey I'm new to this but i figured it should be straightforward enough as in paste key and provide api endpoint.well,I have working keys for openai ,llama and mistral 3b.none seem to want to work like theres some error somewhere..
![Screenshot_20241120-193638_GPTMobile](https://github.com/user-attachments/assets/f5b1bad7-f62b-4188-8146-1717c18545f5)
![Screenshot_20241120-192711_GPTMobile](https://github.com/user-attachments/assets/01c8e249-8a18-49c3-8997-349d0391d301)

**Reply by 4kenhama** [reply]
My api url were set to
/https://api.openai.com/
/https://api.mistral.ai/
/https:api.llama-ai.com/

**Reply by Taewan-P** [reply]
You are missing `v1` in your URL.

**Reply by Walcam** [reply]
So all the url's should end with v1/ or just for openai

**Reply by Taewan-P** [reply]
if the api endpoint is OpenAI compatible and its address contains `v1/`, it should end with `v1/`

ex) API Address `https://api.openai.com/v1/chat/completions` -> Enter `https://api.openai.com/v1/` in address

**Comment by Taewan-P** [comment]
Hello, this feature is now implemented in [v0.7.0](https://github.com/Taewan-P/gpt_mobile/releases). Thanks!
cc @4kenhama @Walcam @MH140000
